x-pack/plugins/enterprise_search/common/types/app_search.ts

```diff
@@ -6,21 +6,18 @@
 
 export interface IAccount {
   accountId: string;
-  onBoardingComplete: boolean;
-  role: IRole;
-}
-
-export interface IRole {
-  id: string;
-  roleType: string;
-  ability: {
-    accessAllEngines: boolean;
-    destroy: string[];
-    manage: string[];
-    edit: string[];
-    view: string[];
-    credentialTypes: string[];
-    availableRoleTypes: string[];
+  onboardingComplete: boolean;
+  role: {
```

Comment:
Haha, I knew I should have left a comment/documented this decision 😄 

I went a bit back and forth on this as well, but in the end I changed it from being its own separate interface because I didn't want people accidentally importing it when they should be using the new `IRole`/`myRole`.

I really wanted to try and emphasize that `myRole` is the data/value we should prefer to use on the client-side - the `account.role` data we're receiving from the server is still there/being used to generate `myRole`, but we generally shouldn't use it.

Hopefully that decision makes sense w/ that context 🤔